Add hot-column, hot-editor and hot-renderer.

jansiegel opened the pull request:
### Context
Add `<hot-column>` subcomponent support, as well as `hot-renderer` and `hot-editor` props.

Sample configuration with custom editors, renderers and additional settings passed in the `hot-column` element:
```vue
<hot-table>
    <hot-column title="Custom Header Title" readOnly="true">
        <custom-renderer hot-renderer></custom-renderer>
        <custom-editor hot-editor></custom-editor>
    </hot-column>
</hot-table>
```
Where:
* `<hot-column>` contains `title` and `readOnly` properties, which are passed to the column configuration in Handsontable
* `<custom-renderer>` is a renderer component with a `hot-renderer` prop added, to mark it as a custom renderer
* `<custom-editor>` is a renderer component with a `hot-editor` prop added, to mark it as a custom editor

### Sample renderer component:
```vue
<template>
  <div>{{ value }}</div>
</template>

<script>
  const CustomRenderer = {
    name: 'CustomCellRenderer',
    data: function() {
      return {
        value: null
      }
    }
  };

  export default CustomRenderer;
</script>
```

### Sample editor component:
```vue
<template>
  <div id="editorContainer">
    <h4>Previous value: {{ value }}</h4>
    <strong>Change to:</strong>
    <button v-on:click="setReversed" >{{ value.split('').reverse().join('') }}</button>
  </div>
</template>

<script>
  const CustomEditor = {
    name: 'CustomCellEditor',
    data: function() {
      return {
        value: ''
      };
    },
    methods: {
      init: function() {
        this.$el.addEventListener('mousedown', event => {
          event.stopPropagation()
        });
      },
      prepare: function(row, col, prop, td, originalValue, cellProperties) {
        if (!document.body.contains(this.$el)) {
          document.body.appendChild(this.$el);
        }

        const tdPosition = td.getBoundingClientRect();

        this.$el.style.left = tdPosition.left + 'px';
        this.$el.style.top = tdPosition.top + 'px';
      },
      setReversed: function() {
        this.setValue(this.value.split('').reverse().join(''));
        this.finishEditing();
      },
      getValue: function() {
        return this.value;
      },
      setValue: function(value) {
        this.value = value;
      },
      open: function() {
        this.$el.style.display = 'block';
      },
      close: function() {
        this.$el.style.display = 'none';
      }
    }
  };

  export default CustomEditor;
</script>

<style scoped>
  div[id="editorContainer"] {
    display: none;
    position: absolute;
    left: 0;
    top: 0;
    background: #fff;
    border: 1px solid #000;
    z-index: 999999;
  }
</style>
```

--------

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Related issue(s):
1. #101

jansiegel committed:
Add the HotColumn component to be used as a child of HotTable to define the column settings. [WIP]

jansiegel committed:
Fill missing typings.

jansiegel committed:
Update package-lock.

jansiegel committed:
Add additional tests.

jansiegel committed:
Add $router and $store tests + move cache to the HotTable class.

jansiegel committed:
Fix tests

jansiegel committed:
Merge branch 'develop' into feature/hotColumn

jansiegel committed:
Specify additional types.

wojciechczerniak reviewed (commented):
Tests are console logging a warning about missing license key. Maybe we should use `non-commercial-and-evaluation`? It was created for testing purposes

jansiegel committed:
Prevent throwing the license key warning in the tests.

jansiegel committed:
Change the renderer cache type to LRU Map, add tests for the renderer cache.

jansiegel committed:
Update parts of the test cases to use '@vue/test-utils'

jansiegel committed:
Fix a problem with HotColumn overwriting the settings passed to HotTable.

jansiegel committed:
Add 'Symbol' as a possible default value for the 'settings' prop + prevent the test cases from throwing warnings about auto column/row size.

jansiegel committed:
Remove misleading 'settings' and hook type declarations.

jansiegel committed:
Don't re-render the TD's contents if it's already rendererd in the viewport.

jansiegel committed:
Add a base editor component for vue components to extend.

jansiegel committed:
Rewrite the BaseVueHotEditor to a class.

jansiegel committed:
Fix tests.

jansiegel committed:
Update package-lock.

jansiegel committed:
Make hot-column inject `isRenderer` and `isEditor` properties to the renderer/editor components.

jansiegel committed:
Add a possibility to declare global editors/renderers as components.

jansiegel committed:
Add a possibility do set the column settings using a 'settings' prop on 'hot-column'.

jansiegel committed:
Change where isEditor prop is injected, correct tests etc.

jansiegel committed:
Correct typo.

jansiegel committed:
Bump the dependencies to the latest versions.

jansiegel committed:
Post CR updates:
- rename BaseVueHotEditor to BaseEditorComponent
- make editor components mount into #vueHotEditors element in DOM
- destroy the initial global editors/renderers

jansiegel committed:
Rename the editor container to `vueHotComponents`, because both editors and renderers are being mounter there (the renderers are later moved to the table cells).

jansiegel committed:
Remove the unneeded store and router rewriting and fix npm audit errors.

jansiegel committed:
Remove some of the querySelector logic and additional cleanup.

jansiegel committed:
Post CR updates.
